daemon/pool.go

```diff
@@ -475,3 +771,31 @@ func (p *aimd) Scale(total, load int) int {
 
 	return res
 }
+
+// chanContext wraps a receive-only channel to act as a context.
```

Comment:
Please let's try to find a solution that doesn't require us to re-implement `context.Context` here. If we really need a separate context, we should—if at all possible—use the one from the `context` package. Instead of creating our own channel.

Can we use:

```go
ctx, cancel := context.WithCancel(context.Background())
```

to get a fresh context, and cache those values for use by the manager?  That gives us to cancel and check for cancellation, and would make it easy to plumb in deadlines if we needed to later.

The Go team strongly discourages re-implementing this interface (it's only an interface because there are different concrete wrappers in its own package for handling the wrapper semantics).

Reply:
Haven't known that reimplementing it is discouraged. Will use `Background` instead.